### PR TITLE
Only handle NXDOMAIN as failure when nameserver is authoritive or no …

### PR DESCRIPTION
**Pull request by normanmaurer**
…other nameservers are left.

Motivation:

When using multiple nameservers and a nameserver respond with NXDOMAIN we should only fail the query if the nameserver in question is authoritive or no nameservers are left to try.

Modifications:

- Try next nameserver if NXDOMAIN was returned but the nameserver is not authoritive
- Adjust testcase to respect correct behaviour.

Result:

Fixes https://github.com/netty/netty/issues/8261

### CODE REVIEW

**Comment by normanmaurer** [comment]
Special thanks to @Lukasa for discussion on the topic :)

**Comment by mrniko** [comment]
I confirm that it fixes https://github.com/netty/netty/issues/8261

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: resolver-dns/src/test/java/io/netty/resolver/dns/DnsNameResolverTest.java [hunk 1]
```diff
@@ -1143,7 +1143,7 @@ public void aAndAAAAQueryShouldTryFirstDnsServerBeforeSecond() throws IOExceptio
                     new TestRecursiveCacheDnsQueryLifecycleObserverFactory();
 
             DnsNameResolverBuilder builder = new DnsNameResolverBuilder(group.next())
-                    .resolvedAddressTypes(ResolvedAddressTypes.IPV6_PREFERRED)
+                    .resolvedAddressTypes(ResolvedAddressTypes.IPV4_ONLY)
```
Comment (ejona86) [inline resolver-dns/src/test/java/io/netty/resolver/dns/DnsNameResolverTest.java]: Why this change?

Comment (normanmaurer) [reply resolver-dns/src/test/java/io/netty/resolver/dns/DnsNameResolverTest.java]: @ejona86 it made the test easier to adjust as it also change the number of queries :)

Comment (ejona86) [reply resolver-dns/src/test/java/io/netty/resolver/dns/DnsNameResolverTest.java]: Ah. Makes sense.

**Comment by normanmaurer** [comment]
@netty-bot test this please
